### PR TITLE
Comparison mode for Qemu log viewer

### PR DESCRIPTION
**Pull request by laerreal**
### v2
- fixup sharing of trace record (CPU, etc.) between instructions in tree view
- outline text styles as `tuple`s
- work on comments

### CODE REVIEW

File: qlv.py [hunk 1]
```diff
@@ -541,14 +549,53 @@ def feed_file_by_name(self, file_name):
...
 
     def co_trace_builder():
         idx = 0
 
         while True:
+            start_idx = idx
             end_idx = idx + 100
-            for idx, i in zip(xrange(idx, end_idx), trace_iter):
+
+            # Build subtrace for first long and then try to compare it with
```
Comment (BDanAnd) [inline qlv.py]: first log

File: qlv.py [hunk 2]
```diff
@@ -645,6 +664,69 @@ def co_trace_builder():
...
+
+        for qlog_idx, (qlog_instrs, trace_text) in enumerate(izip(
+            all_instructions, qlog_trace_texts
+        )):
+            try:
+                i = qlog_instrs[idx]
+            except IndexError:
+                continue
+
+            trace_text.insert(END, args.qlog[qlog_idx] + "\n", ["file"])
```
Comment (BDanAnd) [inline qlv.py]: ```("file",)```

File: qlv.py [hunk 3]
```diff
@@ -645,6 +664,69 @@ def co_trace_builder():
...
+
+            if i.trace is None:
+                trace_text.insert(END, _("No CPU data").get() + "\n",
+                    ["warning"]
```
Comment (BDanAnd) [inline qlv.py]: ```("warning",)```

File: qlv.py [hunk 4]
```diff
@@ -645,6 +664,69 @@ def co_trace_builder():
...
+                )
+            else:
+                trace_text.insert(END, i.trace.as_text)
+
+    tv.bind("<<TreeviewSelect>>", on_instruction_selected, "+")
+
+    # launch trace building (and comparison)
```
Comment (BDanAnd) [inline qlv.py]: Maybe `Launch` for uniformity?

**Comment by laerreal** [comment]
v2 is ready!

File: qlv.py [hunk 5]
```diff
@@ -71,6 +71,29 @@ def __str__(self):
...
+    __slots__ = (
+        "in_instr",
+        "trace",
+    )
+
+    def __init__(self, in_instr, trace):
+        self.in_instr = in_instr
+        self.trace = trace
+
+    # Proxify static info.
```
Comment (BDanAnd) [inline qlv.py:99]: "multitran" does not know the word "proxify".

**Pull request merged**